swarm.go

```diff
@@ -17,6 +17,13 @@ type SwarmAPI HttpApi
 func (api *SwarmAPI) Connect(ctx context.Context, pi peerstore.PeerInfo) error {
 	saddrs := make([]string, len(pi.Addrs))
 	for i, addr := range pi.Addrs {
+		if _, err := addr.ValueForProtocol(multiaddr.P_P2P); err == multiaddr.ErrProtocolNotFound {
```

Comment:
This is going to stumble on p2p-circuit addresses. Really, addresses in `pi.Addrs` shouldn't ed with `/p2p/Qm...` (although they did for a short period of time due to relays, see: https://github.com/libp2p/specs/issues/72).